### PR TITLE
Standardize naming in WellKnownTypeNames

### PR DESCRIPTION
**Pull request by Evangelink**
Relates to #3868

### CODE REVIEW

File: src/Utilities/Compiler/WellKnownTypeNames.cs [hunk 1]
```diff
@@ -252,7 +252,7 @@ internal static class WellKnownTypeNames
         public const string SystemThreadingCancellationToken = "System.Threading.CancellationToken";
         public const string SystemThreadingInterlocked = "System.Threading.Interlocked";
         public const string SystemThreadingMonitor = "System.Threading.Monitor";
-        public const string SystemThreadingTasksGenericTask = "System.Threading.Tasks.Task`1";
+        public const string SystemThreadingTasksTasks = "System.Threading.Tasks.Task`1";
```
Comment (mavasani) [inline src/Utilities/Compiler/WellKnownTypeNames.cs:255]: This should be `SystemThreadingTasksTask1`
```suggestion
        public const string SystemThreadingTasksTask1 = "System.Threading.Tasks.Task`1";
```

**Comment by mavasani** [comment]
@Evangelink I think you would need to rename even the use sites :-)

File: src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotDirectlyAwaitATask.cs [hunk 2]
```diff
@@ -89,7 +89,7 @@ private static void AnalyzeOperation(OperationAnalysisContext context, Immutable
         private static bool TryGetTaskTypes(Compilation compilation, out ImmutableArray<INamedTypeSymbol> taskTypes)
         {
             INamedTypeSymbol? taskType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksTask);
-            INamedTypeSymbol? taskOfTType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksGenericTask);
+            INamedTypeSymbol? taskOfTType = compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksTasks);
```
Comment (sharwell) [inline src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotDirectlyAwaitATask.cs:92]: 😕 I don't think this one is right

**Comment by Evangelink** [comment]
Sorry guys, I have blindly applied the rename suggestion! It's alright now.

**Pull request merged**